### PR TITLE
re-create hadoop's varint

### PR DESCRIPTION
**Pull request by dt**

### CODE REVIEW

**Comment by dt** [comment]
cc @surajacharya

**Comment by dt** [comment]
Hm, I wanted @paperstreet to get a chance to look this over too before we ship it.

File: hfile/vint_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	x, l = vintAndLen(fromHex(t, "10"))
+	assert.Equal(t, 0x10, x)
+	assert.Equal(t, 1, l)
+
+	x, l = vintAndLen(fromHex(t, "a0"))
+	assert.Equal(t, -96, x)
+	assert.Equal(t, 1, l)
+
+	x, l = vintAndLen(fromHex(t, "a0"))
```
Comment (danhhz) [inline hfile/vint_test.go:37]: why is a0 here three times?

Comment (danhhz) [inline hfile/vint_test.go:37]: i'd love to see tests for stuff starting with 0x80 0x87 0x88 0x8f 0x90 0xff 0x00 0x7f just to document all the edges

**Pull request merged**